### DISCUSSION TITLE
Incorporating full dataset into a previous run emcee distribution based on binned dataset

### DISCUSSION THREAD

**Discussion by mami-deka** [phoebe-project/phoebe2#718] [Questions]
Hi!
For fast convergence, I have started the emcee sampling of the parameters using binned light curve and then saved the bundle. Is it possible to replace the binned light curve into full dataset light curve in the bundle?
When I was trying this, I got the following error:

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-14-70ad0073c2f0> in <module>
      1 #print(b.get_parameter(qualifier='init_from', solver='emcee_solver'))
----> 2 b.set_value('init_from', 'emcee_posteriors1')
      3 b.set_value('priors@emcee_solver@emcee@solver', 'emcee_posteriors1')
      4 #b.set_value('nwalkers', solver='emcee_solver', value=36)
      5 #b.set_value('niters', solver='emcee_solver', value=20)

~/anaconda3/lib/python3.8/site-packages/phoebe/parameters/parameters.py in set_value(self, twig, value, **kwargs)
   3391                 return param.set_index_value(value=value, index=index, **kwargs)
   3392 
-> 3393         return self.get_parameter(twig=twig,
   3394                                   **kwargs).set_value(value=value,
   3395                                                       **kwargs)

~/anaconda3/lib/python3.8/site-packages/phoebe/parameters/parameters.py in get_parameter(self, twig, **kwargs)
   3011             matching the search.
   3012         """
-> 3013         return self.get(twig=twig, **kwargs)
   3014 
   3015     def get_or_create(self, qualifier, new_parameter, attach_to_bundle=False, **kwargs):

~/anaconda3/lib/python3.8/site-packages/phoebe/parameters/parameters.py in get(self, twig, check_visible, check_default, check_advanced, check_single, **kwargs)
   2658         if not len(ps):
   2659             # TODO: custom exception?
-> 2660             raise ValueError("0 results found for twig: '{}', {}".format(twig, kwargs))
   2661         elif len(ps) != 1:
   2662             # TODO: custom exception?

ValueError: 0 results found for twig: 'init_from', {'check_visible': True, 'check_default': True, 'check_advanced': False, 'check_single': False}

Please assist me here.

**Comment by kecnry** [comment]
Are you creating a new bundle and trying to reference the previous results?  Or just swapping out the times/fluxes in the dataset in the existing bundle?

**Comment by mami-deka** [comment]
I am trying swapp the time/flux in the previous bundle.

**Reply by kecnry** [reply]
Did you delete the solver options by any chance?  What does `print(b.filter(qualifier='init_from'))` return?

**Comment by mami-deka** [comment]
print(b.filter(qualifier='init_from')) returns the following:

ParameterSet: 0 parameters
NO PARAMETERS

**Reply by kecnry** [reply]
Are your emcee solver options still there - what is the output from `print(b.filter(solver='emcee_solver'))` (or whatever they are named)?  If you set `continue_from`, then `init_from` will be hidden, so you will need to pick one or the other (see the [continue_from](http://phoebe-project.org/docs/latest/tutorials/emcee_continue_from) vs [resampling](http://phoebe-project.org/docs/latest/tutorials/emcee_resampling) tutorials).  If that doesn't help, please attach a minimal script to reproduce so that we can see more context.

**Reply by mami-deka** [reply]
Because of limited computational resource time, I have to run 10000 mcmc runs in parts of 2000 for 5 times using continue_from option.
If I try to swapp the time/flux in the first bundle, everything works fine. But the problem arise when I try to implement this on continued bundle. 

Now I am getting the following error after entering the following line:
b.run_solver('emcee_solver', solution='emcee_sol_full',progress_every_niters=2)

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/anaconda3/lib/python3.8/site-packages/phoebe/solverbackends/solverbackends.py in _fill_solution(self, solution_ps, rpacketlists_per_worker, metawargs)
    510                     try:
--> 511                         solution_ps.set_value(check_visible=False, check_default=False, ignore_readonly=True, **packet)
    512                     except Exception as err:

~/anaconda3/lib/python3.8/site-packages/phoebe/parameters/parameters.py in set_value(self, twig, value, **kwargs)
   3392 
-> 3393         return self.get_parameter(twig=twig,
   3394                                   **kwargs).set_value(value=value,

~/anaconda3/lib/python3.8/site-packages/phoebe/parameters/parameters.py in _send_if_client(self, *args, **kwargs)
    423         else:
--> 424             return fctn(self, *args, **kwargs)
    425 

~/anaconda3/lib/python3.8/site-packages/phoebe/parameters/parameters.py in set_value(self, value, run_checks, **kwargs)
   8374         if not isinstance(value, list):
-> 8375             raise TypeError("value must be a list of strings, received {}".format(type(value)))
   8376 

TypeError: value must be a list of strings, received <class 'numpy.ndarray'>

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-22-6cd7b5cd9175> in <module>
----> 1 b.run_solver('emcee_solver', solution='emcee_sol_full',progress_every_niters=2)

~/anaconda3/lib/python3.8/site-packages/phoebe/parameters/parameters.py in _send_if_client(self, *args, **kwargs)
    422 
    423         else:
--> 424             return fctn(self, *args, **kwargs)
    425 
    426     return _send_if_client

~/anaconda3/lib/python3.8/site-packages/phoebe/frontend/bundle.py in run_solver(self, solver, solution, detach, return_changes, **kwargs)
  13647 
  13648         solver_class = getattr(_solverbackends, '{}Backend'.format(solver_ps.kind.title()))
> 13649         params = solver_class().run(self, solver_ps.solver, compute, solution=solution, **{k:v for k,v in kwargs.items() if k not in ['compute']})
  13650         metawargs = {'context': 'solution',
  13651                      'solver': solver_ps.solver,

~/anaconda3/lib/python3.8/site-packages/phoebe/solverbackends/solverbackends.py in run(self, b, solver, compute, **kwargs)
    553 
    554         else:
--> 555             rpacketlists_per_worker = [self.run_worker(**packet)]
    556 
    557         logger.debug("rank:{}/{} calling _fill_solution".format(mpi.myrank, mpi.nprocs))

~/anaconda3/lib/python3.8/site-packages/phoebe/solverbackends/solverbackends.py in run_worker(self, b, solver, compute, **kwargs)
   1695                         logger.info("emcee: saving output from iteration {}".format(sampler.iteration))
   1696 
-> 1697                         solution_ps = self._fill_solution(solution_ps, [_get_packetlist()], metawargs)
   1698 
   1699                         if 'out_fname' in kwargs.keys():

~/anaconda3/lib/python3.8/site-packages/phoebe/solverbackends/solverbackends.py in _fill_solution(self, solution_ps, rpacketlists_per_worker, metawargs)
    511                         solution_ps.set_value(check_visible=False, check_default=False, ignore_readonly=True, **packet)
    512                     except Exception as err:
--> 513                         raise ValueError("failed to set value from packet: {}.  Original error: {}".format(packet, str(err)))
    514 
    515         if metawargs:

ValueError: failed to set value from packet: {'qualifier': 'adopt_parameters', 'value': array(['teffratio@binary@orbit@component', 'teff@primary@star@component',
       'requivsumfrac@binary@orbit@component',
       'requivratio@binary@orbit@component',
       'incl@binary@orbit@component', 'sma@binary@orbit@component',
       'ecc@binary@orbit@component', 'per0@binary@orbit@component',
       'pblum@primary@lc05@lc@dataset',
       'gravb_bol@primary@star@component',
       'gravb_bol@secondary@star@component',
       'irrad_frac_refl_bol@primary@star@component',
       'irrad_frac_refl_bol@secondary@star@component',
       'sigmas_lnf@lc05@lc@dataset'], dtype='<U44')}.  Original error: value must be a list of strings, received <class 'numpy.ndarray'>

**Reply by kecnry** [reply]
This might be a bug, but it's very hard to tell without more context or an example to reproduce.  Can you send at least the solver options for this latest attempt - `print(b.filter(solver='emcee_solver'))`?

**Reply by mami-deka** [reply]
Thank you for your patience with me. 

However, I have attached the result of the following statement below:
print(b.filter(solver='emcee_solver', context='solver'))

ParameterSet: 11 parameters
     comments@emcee_solver@solver: 
   use_server@emcee_solver@solver: compute
      compute@emcee_solver@solver: phoebe01
   continue_from@emcee_solver@...: emcee_sol_contd
   continue_from_iter@emcee_so...: -1
       priors@emcee_solver@solver: ['ball_around_guess']
       niters@emcee_solver@solver: 2000
   burnin_factor@emcee_solver@...: 2.0
   thin_factor@emcee_solver@so...: 0.0
   nlags_factor@emcee_solver@s...: 3.0
   progress_every_niters@emcee...: 0

**Reply by mami-deka** [reply]
Please find the script in the following link.
https://drive.google.com/file/d/1ODF4vXkxAtOIrdkoqqmKEM_BN8vg2Vp0/view?usp=sharing

**Reply by kecnry** [reply]
I cannot figure out where in the code this was turning into an array for you and the script/bundle was too large to try to reproduce, but I opened #720 which should avoid this error by allowing it to be passed internally as an array.  That will hopefully be released in version 2.4.10 within the next few days.

**Reply by mami-deka** [reply]
Thank you for your time and effort!
